interp/src/flatten/structures/indexed_map.rs

```diff
@@ -143,6 +143,20 @@ where
         Self::new()
     }
 }
+
+impl<T, K> Clone for IndexedMap<K, T>
+where
+    K: IndexRef,
+    T: Clone,
+{
+    fn clone(&self) -> Self {
+        Self {
+            data: self.data.clone(),
+            phantom: PhantomData,
+        }
+    }
+}
+
```

Comment:
Does `#[derive(Clone)]` work with generic bounds?

Reply:
Related to the `derivative` crate -- I could add that crate to make a clean
```
#[derive(Derivative)]
#[derivative(Clone(bound="K: IndexRef, T: Clone"))
```
but it would be the only such usage.

Reply:
`#[derive(Clone)]` does work with generic bounds.

See: https://stackoverflow.com/questions/72150623/deriveclone-seems-to-wrongfully-enforce-generic-to-be-clone/72150746#72150746

Reply:
(This is fresh in my head since it's also something I just realized recently)